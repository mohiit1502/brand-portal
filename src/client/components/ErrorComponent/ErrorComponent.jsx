import React from "react";
import PropTypes from "prop-types";
import "./ErrorComponent.component.scss";

const ErrorComponent = props => {
  return props.error ? <div className={`c-ErrorComponent${props.containerClasses ? ` ${  props.containerClasses}` : ""}`}>
      <small className={`${props.errorClasses}`}>{ props.error }</small>
    </div> : null;
};

ErrorComponent.propTypes = {
  containerClasses: PropTypes.string,
  errorClasses: PropTypes.string,
  error: PropTypes.string
};

export default ErrorComponent;
