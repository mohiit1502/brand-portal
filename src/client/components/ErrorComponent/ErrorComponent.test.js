import React, {useRef} from "react";
import ErrorComponent from "./ErrorComponent";
import toJson from "enzyme-to-json";
import {mount} from "enzyme";
import moment from "moment";
import PropTypes from "prop-types";

let props;

const setUp = (props) => {
  return mount(props ? <ErrorComponent {...props} /> : <ErrorComponent />);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});



describe("ErrorComponent", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("renders without error", () => {
    props = {
      containerClasses: "",
      errorClasses: "",
      error: ""
    };
    wrapper = setUp(props);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

  });
});
