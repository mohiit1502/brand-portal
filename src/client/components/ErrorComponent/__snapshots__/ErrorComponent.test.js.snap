// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent renders without error 1`] = `
<ErrorComponent
  containerClasses=""
  error=""
  errorClasses=""
/>
`;
